server/src/main/java/org/apache/druid/client/CachingClusteredClient.java

```diff
@@ -844,23 +845,26 @@ private byte[] computeQueryCacheKeyWithJoin()
     }
   }
 
-  private static class TimelineConverter implements UnaryOperator<TimelineLookup<String, ServerSelector>>
+  /**
+   * Helper class to build a new timeline of filtered segments.
+   */
+  public static class TimelineConverter<ObjectType extends Overshadowable<ObjectType>> implements UnaryOperator<TimelineLookup<String, ObjectType>>
```

Comment:
nit: is there really nothing more descriptive than `ObjectType`? Might as well just use `T` if not 😅